Sources/WeaverCodeGen/SwiftGenerator.swift

```diff
@@ -1305,6 +1334,109 @@ private extension MetaWeaverFile {
         }
     }
 
+    func providerClassAndFunctionsMainActor() throws -> [FileBodyMember] {
```

Comment:
If I understand correctly this method is a copy of
`func providerClassAndFunctions() throws -> [FileBodyMember]` with `@MainActor` annotations added where needed.

What do you think about modifying `func providerClassAndFunctions() throws -> [FileBodyMember]` by putting `\(self.mainactor ? "@MainActor " : "")` in appropriate places instead of introducing new method?

This seems to be safer solution, because if in the future one of the methods needs to be modified, the change would have to occur only in one place instead of two.

Reply:
I ended up breaking this out into a separate method as other changes were initially going into that string, not just additions of "@ MainActor".  At the moment, `\(self.mainactor ? "@MainActor " : "")` would suffice, but I kind of suspect that more significant changes will want to be made to this in the future.